package.json

```diff
@@ -5,20 +5,41 @@
   "scripts": {
     "dev": "next dev src/ui",
     "build": "next build src/ui",
-    "lint": "next lint src/ui"
+    "lint": "next lint src/ui",
+    "type-check": "tsc -p src/ui/tsconfig.json --noEmit",
+    "format": "prettier --ignore-path src/ui/.prettierignore --write \"src/ui/**/*.+(js|ts|jsx|tsx|json)\"",
+    "prepare": "husky install"
   },
   "dependencies": {
+    "next": "15.3.2",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0",
-    "next": "15.3.2"
+    "react-dom": "^18.2.0"
   },
   "devDependencies": {
-    "typescript": "^5",
+    "@eslint/eslintrc": "^3",
     "@types/node": "^20",
     "@types/react": "^18.0.28",
     "@types/react-dom": "^18.0.11",
-    "eslint": "^9",
+    "@typescript-eslint/eslint-plugin": "^5.43.0",
+    "@typescript-eslint/parser": "^5.43.0",
+    "eslint": "^8.27.0",
     "eslint-config-next": "15.3.2",
-    "@eslint/eslintrc": "^3"
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jsx-a11y": "^6.6.1",
+    "eslint-plugin-no-secrets": "^1.0.2",
+    "eslint-plugin-prettier": "^5.4.0",
+    "eslint-plugin-react": "^7.31.10",
+    "eslint-plugin-react-hooks": "^5.2.0",
+    "husky": "^9.1.7",
+    "prettier": "^3.5.3",
+    "tslint-config-prettier": "^1.18.0",
+    "typescript": "^5"
+  },
+  "lint-staged": {
+    "*.js": "eslint --cache --fix",
+    "*.ts": "eslint --cache --fix",
+    "*.jsx": "eslint --cache --fix",
+    "*.tsx": "eslint --cache --fix"
   }
 }
```

Comment:
Can we add engines in here as well to standardize the node version someone is using? IE:
```
"engines": {
  "node": ">=X.X"
}
```

For the version, 22 looks like it would be a good one since it's flagged as LTS and will be maintained for awhile: https://nodejs.org/en/about/previous-releases